config/webpack/environment.js

```diff
@@ -1,25 +1,35 @@
+/* global require module */
 const { environment } = require('@rails/webpacker');
 const erb = require('./loaders/erb');
-const webpack = require('webpack');
 
-environment.plugins.append(
-  'CommonsChunkVendor',
-  new webpack.optimize.CommonsChunkPlugin({
-    name: 'vendor',
-    minChunks: module => {
-      // this assumes your vendor imports exist in the node_modules directory
-      return module.context && module.context.indexOf('node_modules') !== -1;
+environment.splitChunks(config => {
```

Comment:
@nickytonline can you tell us more about this configuration?

Reply:
This just to create the vendor chunk. The vendor chunk is not like it was previously. There is no longer one vendor bundle. It gets code split based on what webpacker packs need.

All the object spreading e.g. `...config.optimization` is to keep the existing configuration and only override/add what is necessary. The cache groups section I have here is the [default cache groups in webpack 4](https://webpack.js.org/plugins/split-chunks-plugin/#optimizationsplitchunks), but it does not appear to be the default with webpacker 4.

Reply:
feel free to add this explanation as a comment :D

Reply:
Will do. 😄